src/lib/operators/table_scan.cpp

```diff
@@ -300,8 +300,8 @@ std::unique_ptr<AbstractTableScanImpl> TableScan::create_impl() const {
 
     auto lower_bound_value = expression_get_value_or_parameter(*between_expression->lower_bound());
     if (lower_bound_value) {
-      const auto adjusted_predicate_and_value =
-          lossless_predicate_variant_cast(lower_condition, *lower_bound_value, left_column->data_type());
+      const auto adjusted_predicate_and_value = lossless_predicate_variant_cast(
+          lower_condition, *lower_bound_value, between_expression->value()->data_type());
       if (adjusted_predicate_and_value) {
         lower_condition = adjusted_predicate_and_value->first;
         lower_bound_value = adjusted_predicate_and_value->second;
```

Comment:
I know BETWEEN statements, but this implementation seemed Greek to me. 

1. Split up predicate:`auto [lower_condition, upper_condition] = between_to_conditions(predicate_condition);`
2. Some stuff -> **please comment**
3. Bringt it back together again: `conditions_to_between(lower_condition, upper_condition)`